crates/uv-requirements/src/specification.rs

```diff
@@ -129,70 +128,11 @@ impl RequirementsSpecification {
-                // 621 file, but might still resolve under PEP 517. (If the source tree doesn't
-                // resolve under PEP 517, we'll catch that later.)
-                //
-                // For example, Hatch's "Context formatting" API is not compliant with PEP 621, as
-                // it expects dynamic processing by the build backend for the static metadata
-                // fields. See: https://hatch.pypa.io/latest/config/context/
-                if let Some(project) = pyproject
-                    .project
-                    .and_then(|project| Pep621Metadata::try_from(project, extras).ok().flatten())
-                {
-                    Self {
-                        project: Some(project.name),
-                        requirements: project
-                            .requirements
-                            .into_iter()
-                            .map(|requirement| RequirementEntry {
-                                requirement: RequirementsTxtRequirement::Pep508(requirement),
-                                hashes: vec![],
-                            })
-                            .collect(),
-                        constraints: vec![],
-                        overrides: vec![],
-                        editables: vec![],
-                        source_trees: vec![],
-                        extras: project.used_extras,
-                        index_url: None,
-                        extra_index_urls: vec![],
-                        no_index: false,
-                        find_links: vec![],
-                        no_binary: NoBinary::default(),
-                        no_build: NoBuild::default(),
-                    }
-                } else {
-                    let path = fs_err::canonicalize(path)?;
-                    let source_tree = path.parent().ok_or_else(|| {
-                        anyhow::anyhow!(
-                            "The file `{}` appears to be a `pyproject.toml` file, which must be in a directory",
-                            path.user_display()
-                        )
-                    })?;
-                    Self {
-                        project: None,
-                        requirements: vec![],
-                        constraints: vec![],
-                        overrides: vec![],
-                        editables: vec![],
-                        source_trees: vec![source_tree.to_path_buf()],
-                        extras: FxHashSet::default(),
-                        index_url: None,
-                        extra_index_urls: vec![],
-                        no_index: false,
-                        find_links: vec![],
-                        no_binary: NoBinary::default(),
-                        no_build: NoBuild::default(),
-                    }
-                }
+                // We need to join relative paths inside pyproject.toml to this.
```

Comment:
What does this comment mean? It needs more context, if I can't understand what it's referring to now I definitely won't be able to in the future!